db/compaction_iterator.cc

```diff
@@ -269,6 +279,20 @@ void CompactionIterator::NextFromInput() {
     // compaction filter). ikey_.user_key is pointing to the copy.
     if (!has_current_user_key_ ||
         !cmp_->Equal(ikey_.user_key, current_user_key_)) {
+      key_cnt_++;
+      // Use key_cnt_ to reduce the overhead of reading current time
+      if (snap_list_callback_ && snapshots_->size() &&
+          key_cnt_ % snap_list_callback_->key_skip() == 0) {
```

Comment:
Calling one division per key is something we should avoid. Can we use integer shifting instead here?